src/lazy-loader.marko

```diff
@@ -7,18 +7,24 @@ class {
       componentRenderer: null,
       renderMethod: input.renderMethod || null,
       enhanceModel: input.enhanceModel || null,
+      customLoadEvent: input.customLoadEvent || false,
+      customLoadEventTriggered: false,
       rawHtml:null
     };
   }
   onMount() {
     this.checkForLazyLoads = this.checkForLazyLoads.bind(this);
     this.doCheckForLazyLoads = this.doCheckForLazyLoads.bind(this);
+    this.doCustomLoadEvent = this.doCustomLoadEvent.bind(this);
     window.addEventListener("scroll", this.checkForLazyLoads);
     window.addEventListener("resize", this.checkForLazyLoads);
     window.addEventListener("focus", this.checkForLazyLoads);
     document.addEventListener("checkForLazyLoads", this.checkForLazyLoads)
     if (document.readyState === "complete") this.checkForLazyLoads();
     else window.addEventListener("load", this.checkForLazyLoads); // Makes sure lazy loaded components render immediately if in initial viewport
+
+    //set a custom trigger to bypass checks and just load component
+    if(this.state.customLoadEvent) document.addEventListener(this.state.customLoadEvent, this.doCustomLoadEvent)
```

Comment:
don't need a new function (referring to doCustomLoadEvent) can just directly call the existing loadComponent